### ISSUE TITLE
not working error = 'ethernets'

### ISSUE THREAD

**Issue by onlineapps-cloud** [xinthose/Netplan-GUI#11]
hi, i installed on instruction, and all ok, but i unable to change my ip from dhcp to static
when i click submit i get this:
![image](https://github.com/xinthose/Netplan-GUI/assets/69077240/9475623a-5f8e-4778-abb8-9e9a2efcac46)
pm2 logs:
```
2|netplan-gui-rest  | 2023-11-16 22:19:18,508 [ERROR] rest.py/submitEth1:431 >> error = 'ethernets'
2|netplan-gui-rest  | 2023-11-16 22:19:18,523 [ERROR] rest.py/submitEth1:431 >> error = 'ethernets'
```
netstat -lntup
`tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN      53700/python3.8 `

`Linux ubuntu 6.2.0-36-generic #37~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon Oct  9 15:34:04 UTC 2 x86_64 x86_64 x86_64 GNU/Linux`
```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy
```
```
pip -V
pip 23.3.1 from /usr/local/lib/python3.8/dist-packages/pip (python 3.8)
```

```
python3 -V
Python 3.10.12
```
other logs:

```
2023-11-16 22:31:43,019 [INFO] rest.py/<module>:48 >> netplan gui REST started; version = 1.1.0
2023-11-16 22:31:43,024 [DEBUG] selector_events.py/__init__:59 >> Using selector: EpollSelector
2023-11-16 22:32:50,980 [ERROR] rest.py/submitEth1:431 >> error = 'ethernets'
2023-11-16 22:32:50,999 [ERROR] rest.py/submitEth1:431 >> error = 'ethernets'
```
```
cat /etc/netplan/01-network-manager-all.yaml
# Let NetworkManager manage all devices on this system
network:
  version: 2
  renderer: NetworkManager
```

error log is empty:
![image](https://github.com/xinthose/Netplan-GUI/assets/69077240/f230dd23-0e98-4445-b207-351df4c7ca1a)
i tried withoud and with `apt install python3-fastapi python3-gunicorn python3-simplejson`
same results, which info need for debug? thanks, best regards.

**Comment by xinthose** [comment]
OK.  I see the issue.  In the `submitEth1` function, It is looking for an already existing `ethernets` property that is not there in your generic `01-network-manager-all.yaml` file [[CODE](https://github.com/xinthose/Netplan-GUI/blob/a105f06a82645575dc8d81cfb0b01b0bfca1467c/src/assets/rest/rest.py#L407)].  Please edit your file and put in the contents below.  

```yaml
network:
  ethernets:
    eth0:
      addresses:
      - 192.168.4.31/22
      dhcp4: false
      dhcp6: false
      match:
        macaddress: 24:4b:fe:e2:1c:4a
      nameservers:
        addresses:
        - 8.8.8.8
        - 8.8.4.4
      routes:
      - to: default
        via: 192.168.4.1
      set-name: eth0
  renderer: NetworkManager
  version: 2
```

This will prime the pump and allow the submit to work.

Thank you for reporting this issue, I will fix it in the next release.

**Comment by onlineapps-cloud** [comment]
thanks for answer, also question, why you not use existent MAC address? it's not conveniently get address from machine where i don't have ssh access, maybe you will get it automatically from network? also it's possible to show current network status?, in my case is configured in DHCP mode. to switch betwen DHCP and static configuration, it will be great.

**Comment by xinthose** [comment]
These are all good ideas!  I will see if I can work on them over the holiday weekend.

**Comment by xinthose** [comment]
Issue closed in [Release 1.2.0](https://github.com/xinthose/Netplan-GUI/releases/tag/v1.2.0).
